python/dgl/transform.py

```diff
@@ -520,4 +524,38 @@ def remove_self_loop(g):
     new_g.add_edges(src[non_self_edges_idx], dst[non_self_edges_idx])
     return new_g
 
+def partition_graph_with_halo(g, node_part, num_hops):
```

Comment:
`node_part` contains node assignment to partitions. `partition_graph_with_halo` partitions a graph and generates physical graph structures. `extend_subgraph_with_halo` isn’t right because the input graph is the original graph, not the partitioned subgraph.